contracts/contracts/TACoApplication.sol

```diff
@@ -228,6 +216,26 @@ contract PREApplication is IApplication, Adjudicator, OwnableUpgradeable {
         __Ownable_init();
     }
 
+    /**
+     * @notice Set contract for multi-chain interactions
+     */
+    function setUpdatableStakeInfo(IUpdatableStakeInfo _updatableStakeInfo) external onlyOwner {
+        require(address(_updatableStakeInfo) != address(updatableStakeInfo), "New address must not be equal to the current one");
+        if (address(_updatableStakeInfo) != address(0)) {
+            // trying to call contract to be sure that is correct address
+            _updatableStakeInfo.updateOperator(address(0), address(0));
```

Comment:
So this is to implicitly check the `source` in `PolygonRoot`, right?

Reply:
and to check that contract is correct one in general